### PR TITLE
fix: make redirection screen more user friendly

### LINKED ISSUE #484: Redirect without a `url` hangs on the redirection screen

### PR DESCRIPTION
**Pull request by spaenleh**
When the user comes from `graasp.org` the `url` query param is not set. 
When the user goes to `auth.graasp.org` the backend does not provide a url to go to and the redirection screen does not know where to send the user. 

To fix this issue:
- change the redirection screen to make it less like "You have to wait, something will happen"
- add a fixed fallback env value that goes to `builder.graasp.org` in each environnement for when there is nothing

Current Screen:

Screenshot 2024-10-22 at 08 57 30

New Screen:

Screenshot 2024-10-22 at 10 24 19

close #484

### CODE REVIEW

File: src/langs/en.json [hunk 1]
```diff
@@ -82,5 +82,8 @@
   "RESET_PASSWORD_REQUIREMENTS_NUMBER": "Contain at least {{count}} number",
   "SHOW_PASSWORD": "Show passwords",
   "INVALID_TOKEN_PROVIDED_TITLE": "Invalid token provided",
-  "INVALID_TOKEN_PROVIDED_DESCRIPTION": "No token was provided or the provided token is expired. Click the button below to generate a request to reset your password."
+  "INVALID_TOKEN_PROVIDED_DESCRIPTION": "No token was provided or the provided token is expired. Click the button below to generate a request to reset your password.",
+  "REDIRECTION_TITLE": "Welcome back, {{name}}",
+  "REDIRECTION_DESCRIPTION": "You should be redirected to your content shortly. You can navigate directly by clicking the button below.",
+  "REDIRECTION_BUTTON": "Go to content"
```
Comment (pyphilia) [inline src/langs/en.json:88]:
```suggestion
  "REDIRECTION_DESCRIPTION": "You should be redirected to Graasp shortly. You can navigate directly by clicking the button below.",
  "REDIRECTION_BUTTON": "Go to Graasp"
```

File: cypress/e2e/SignIn.cy.ts [hunk 2]
```diff
@@ -14,3 +17,16 @@ describe('Name and Email Validation', () => {
...
+
+describe('Already signed in', () => {
+  beforeEach(() => {
+    cy.setUpApi({ currentMember: MEMBERS.BOB });
+  });
+  it('Should show redirection content', () => {
+    cy.visit('/');
+    cy.get(`#${REDIRECTION_CONTENT_CONTAINER_ID}`);
+    cy.get(`[role="button"]`).click();
+    cy.get('h1').should('contain', 'Content');
```
Comment (pyphilia) [inline cypress/e2e/SignIn.cy.ts:29]: Reuse text var?

File: cypress/support/server.ts [hunk 3]
```diff
@@ -82,3 +83,17 @@ export const mockResetPassword = (shouldThrowServerError = false) => {
...
+    {
+      method: HttpMethod.Get,
+      url: DEFAULT_REDIRECTION_URL,
+    },
+    ({ reply }) =>
+      reply({
+        body: '<h1>Content</h1>',
+        headers: { 'content-type': 'text/html' },
+      }),
+  ).as('resetPassword');
```
Comment (pyphilia) [inline cypress/support/server.ts:98]: ?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: cypress/e2e/SignIn.cy.ts (hunk 2 above)
Comment (spaenleh) [reply cypress/e2e/SignIn.cy.ts:29]: This tests the text that is returned by the static mock. Should I add it in a constant ?

File: cypress/support/server.ts (hunk 3 above)
Comment (spaenleh) [reply cypress/support/server.ts:98]: Woops

**Comment by spaenleh** [comment]
@pyphilia I tried with the automatic redirection and I did not really like it. I think it is a bit nicer to have a welcome and click the button. This lets users know that they are logged in. Also this prevents any redirection loops that could be hard to debug if the url param is set to something that makes it look over itself.

**Review by pyphilia — APPROVED** [review]

**Pull request merged**